diff --git a/cmake/QtModuleHelpers.cmake b/cmake/QtModuleHelpers.cmake
index 090e1343b0..143c6d79e5 100644
--- a/cmake/QtModuleHelpers.cmake
+++ b/cmake/QtModuleHelpers.cmake
@@ -286,7 +286,7 @@ function(qt_internal_add_module target)
                                  -outdir "${QT_BUILD_DIR}"
                                  -builddir "${PROJECT_BINARY_DIR}"
                                  "${PROJECT_SOURCE_DIR}")
-        message(STATUS "Running syncqt for module: '${module_include_name}' ")
+        message(STATUS "Running syncqt for module: '${module_include_name}' '${syncqt_full_command}'")
         execute_process(COMMAND ${syncqt_full_command} RESULT_VARIABLE syncqt_ret)
         if(NOT syncqt_ret EQUAL 0)
             message(FATAL_ERROR "Failed to run syncqt, return code: ${syncqt_ret}")
